src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java

```diff
@@ -163,6 +181,12 @@ public int compare(Build b1, Build b2) {
                             .inNamespace(namespace).withName(buildConfigName)
                             .get();
                     if (bc == null) {
+                        if (OpenShiftUtils.isJenkinsBuild(b)) {
```

Comment:
why are we only doing this check here?  Shouldn't the entire loop only look at JenkinsBuilds? (i assume this really means "pipeline strategy builds"?)

Reply:
yeah testing did not show a need, and I went through both pipeline strat and non-pipeline ... there are similar short circuits for pipeline bc's, etc. that may catch it

but that said, it does look inconsistent .... I'll either make the checks more pervasive, or move the gating logic down to addBuildToNoBCList

Reply:
and of course, to be clear, we are going down a thread of optimizing sync plugin in general, not just for the nuances of this new batching mechanism for builds which arrive before their corresponding BCs do

i.e. this should have been done regardless 

it still seems the right thing to do (unless there is some unforeseen side effect I'm not picking up on yet) and i'll proceed